### ISSUE TITLE
Catch and handle schema generation errors/warnings

### ISSUE THREAD

**Issue by JeffJankowski** [graphile/postgraphile#1634]
### Feature description

When instantiating the postgraphile middleware for express there doesn't seem to be any way to catch possibles errors thrown during the schema generation process. Currently any issues are printed to the console, but it would be nice to have a more robust solution.

We are `await`ing `getGraphQLSchema()` before starting the HTTP server, so having this method throw exceptions would be sufficient for our use-case. 

### Motivating example

Our development server uses library mode and is deployed in a cloud environment so watch mode isn't an option for us. Therefore whenever our DBA makes changes that affects the graphql schema, I set up a CI/CD process he can trigger that essentially just restarts the server.

Sometimes he misses a smart tag or something and the server encounters a recoverable error, and he's left wondering why the graphql schema doesn't reflect his changes as expected. I would be great if I could catch the error/warning and notify him directly instead of having to investigate myself 😛 

### Breaking changes

Could be problematic for some if postgraphile starts throwing exceptions for errors that could previously be safely ignored..

### Supporting development

I [tick all that apply]:

- [ ] am interested in building this feature myself
- [x] am interested in collaborating on building this feature
- [x] am willing to help testing this feature before it's released
- [ ] am willing to write a test-driven test suite for this feature (before it exists)
- [ ] am a [Graphile sponsor](https://www.graphile.org/sponsor/) ❤️
- [ ] have an active [support or consultancy contract](https://www.graphile.org/support/) with Graphile

**Comment by benjie** [comment]
Have you looked into the `retryOnInitFail` function? It gets passed any error that causes the server to fail to start.

Another option is to override `SwallowErrorsPlugin` to add your own handling for "swallowed" errors: https://github.com/graphile/graphile-engine/blob/v4/packages/graphile-build/src/plugins/SwallowErrorsPlugin.js

Do one or both of these solve your issue?

**Comment by JeffJankowski** [comment]
Apologies for taking so long to circle back here. Yes, the provided entry points are sufficient for my use case. Thanks as always @benjie for support!
